src/test/java/cd/go/authentication/passwordfile/AuthenticatorTest.java

```diff
@@ -104,6 +104,48 @@ public void shouldAuthenticateBCryptHashedPassword2bFormat() throws Exception {
     }
 
     @Test
+    public void shouldAuthenticatePBKDF2SHA1Format() throws Exception {
+        final String hashed = new StringBuilder()
+                .append("$PBKDF2WithHmacSHA1")
+                .append("$1000")
+                .append("$256")
+                .append("$85A1CCB7C4F4C535CBABDDBD01FECB7B")
+                .append("$fa487eeb17946558c49055e4ecb3faa02a3c847738df67231e5d1a5f33fba7fc").toString();
+
+        Credentials credentials = new Credentials("username", "badger");
+        AuthConfig authConfig = AuthConfigMother.authConfigWith("/var/etc/password.properties");
+        final Properties properties = new Properties();
+
+        properties.put("username", hashed);
+        when(passwordFileReader.read(authConfig.getConfiguration().getPasswordFilePath())).thenReturn(properties);
+
+        final User user = authenticator.authenticate(credentials, Collections.singletonList(authConfig));
+
+        assertThat(user, is(new User("username", "username", null)));
```

Comment:
For the sake of sanity, can we get some negative tests in all cases.

`assertThat(authenticator.authenticate(badCredentials, ...), isNull())`